### ISSUE TITLE
Whitelist directories is too simple

### ISSUE THREAD

**Issue by spiralsam** [keyoti/VSCode-ChangeAllEOL#15]
It looks like you can only whiteslist one level deep.
For example, in a root folder with this structure:

```shell
<root>
-dir1
--subdir1
---sub-subdir1
-dir1
```

Only directory "dir1" or "dir2" can be whitelisted. I've tried. It's impossible to whitelist /dir1/subdir1, much less /dir1/subdir1/sub-subdir1. Unless there's some undocumented, unintuitive way to do it.

**Comment by keyoti** [comment]
Yes, I think it's because the code iterates the current folder and see's each file folder name only and doesn't see the full path
```

for (const [name, type] of await vscode.workspace.fs.readDirectory(folder)) {
...
				if (type === vscode.FileType.Directory
					&& !blackList.includes(name)
					&& (!whiteList.length || whiteList.includes(name))
```

Someone could submit a PR to include a variable that tracks the relative path of the current folder, and then whitelist would be able to work with dir1/subdir1 and that should automatically be backwards compatible.
